pkg/apb/executor.go

```diff
@@ -76,20 +88,49 @@ func ExecuteApb(
 						extraVars,
 					},
 					ImagePullPolicy: pullPolicy,
+					VolumeMounts:    volumeMounts,
 				},
 			},
 			RestartPolicy:      v1.RestartPolicyNever,
-			ServiceAccountName: serviceAccountName,
+			ServiceAccountName: executionContext.ServiceAccount,
+			Volumes:            volumes,
 		},
 	}
 
-	log.Notice(fmt.Sprintf("Creating pod %q in the %s namespace", pod.Name, ns))
+	log.Notice(fmt.Sprintf("Creating pod %q in the %s namespace", pod.Name, executionContext.Namespace))
 	k8scli, err := clients.Kubernetes(log)
 	if err != nil {
-		return apbID, err
+		return executionContext, err
+	}
+	_, err = k8scli.CoreV1().Pods(executionContext.Namespace).Create(pod)
+
+	return executionContext, err
```

Comment:
Actually, some context does need to be returned if it has been created (such as namespace or service account name), so that we can run cleanup on it correctly. 

I could move the cleanup code into the executor module, in which case it would also have be responsible for doing stuff like gathering creds etc. That might actually be better and a little more sensical, since all the code for interacting with apbs would be contained in the executor module, the actions wouldn't need to know how things are done or how to clean up what the executor module has done. It might also make it easier to switch to different cluster types etc in the longer term.